src/components/checkbox-group/checkbox-group.test.ts

```diff
@@ -0,0 +1,298 @@
+import '../../../dist/shoelace.js';
+import { aTimeout, expect, fixture, html, oneEvent, waitUntil } from '@open-wc/testing';
+import { runFormControlBaseTests } from '../../internal/test/form-control-base-tests.js';
+import { sendKeys } from '@web/test-runner-commands';
+import sinon from 'sinon';
+import type { SlChangeEvent } from '../../events/sl-change.js';
+import type SlCheckbox from '../checkbox/checkbox.js';
+import type SlCheckboxGroup from './checkbox-group.js';
+
+describe('<sl-checkbox-group>', () => {
+  describe('validation tests', () => {
+    it('should be invalid initially when required and no checkbox is checked', async () => {
+      const checkboxGroup = await fixture<SlCheckboxGroup>(html`
+        <sl-checkbox-group required>
+          <sl-checkbox value="1"></sl-checkbox>
+          <sl-checkbox value="2"></sl-checkbox>
+        </sl-checkbox-group>
+      `);
+
+      expect(checkboxGroup.checkValidity()).to.be.false;
+    });
+
+    it('should become valid when an option is checked', async () => {
+      const checkboxGroup = await fixture<SlCheckboxGroup>(html`
+        <sl-checkbox-group required>
+          <sl-checkbox value="1"></sl-checkbox>
+          <sl-checkbox value="2"></sl-checkbox>
+        </sl-checkbox-group>
+      `);
+
+      const secondCheckbox = checkboxGroup.querySelectorAll('sl-checkbox')[1];
+      secondCheckbox.click();
+      await checkboxGroup.updateComplete;
+      expect(checkboxGroup.checkValidity()).to.be.true;
+    });
+
+    it(`should be valid when required and one checkbox is checked`, async () => {
+      const el = await fixture<SlCheckboxGroup>(html`
+        <sl-checkbox-group label="Select an option" required>
+          <sl-checkbox name="option" value="1" checked>Option 1</sl-checkbox>
+          <sl-checkbox name="option" value="2">Option 2</sl-checkbox>
+          <sl-checkbox name="option" value="3">Option 3</sl-checkbox>
+        </sl-checkbox-group>
+      `);
+
+      expect(el.checkValidity()).to.be.true;
+    });
+
+    it(`should be invalid when required and no checkboxes are checked`, async () => {
+      const el = await fixture<SlCheckboxGroup>(html`
+        <sl-checkbox-group label="Select an option" required>
+          <sl-checkbox name="option" value="1">Option 1</sl-checkbox>
+          <sl-checkbox name="option" value="2">Option 2</sl-checkbox>
+          <sl-checkbox name="option" value="3">Option 3</sl-checkbox>
+        </sl-checkbox-group>
+      `);
+
+      expect(el.checkValidity()).to.be.false;
+    });
```

Comment:
Is this different than the check on line 12?

Reply:
I see what you mean. I was trying to follow the test patterns for the radio-group for these (these are the un-edited tests from the version of Shoelace we're using), and the `radio-group.test.ts` essentially has the same thing: 
- If you compare the tests at **line 13** vs **line 50** in this file: [src/components/radio-group/radio-group.test.ts](src/components/radio-group/radio-group.test.ts)